web_programming/reddit.py

```diff
@@ -0,0 +1,45 @@
+import praw
+
+credentials = {
+    "client_id": "YOUR CLIENT ID",
+    "client_secret": "YOUR CLIENT SECRET",
+    "user_agent": "A unique name",
+}
+
+
+def get_data(credentials: dict, subreddit: str, limit: int = 1) -> dict:
+    """
+    Credentials should be fetched from https://www.reddit.com/prefs/apps/
+    Create a dict in the format(replace <text> with appropriate data):
+        {
+            "client_id" : <YOUR CLIENT ID>
+            "client_secret" : <YOUR CLIENT SECRET>
+            "user_agent" : <YOUR USER AGENT(Just a unique name)>
+        }
+    The fetched data has following schema:
+        {
+            0: {
+                "title": "SOME TITLE",
+                "url" : "URL OF POST",
+                "content" : "CONTENT",
+            },
+
+            1: {...},
+            2: {...},
+            ...
+        }
+    """
+    reddit = praw.Reddit(**credentials)
+
+    data: dict = {}
+    for _id, submission in enumerate(reddit.subreddit(subreddit).hot(limit=limit)):
```

Comment:
What does `type(reddit.subreddit(subreddit).hot(limit=limit))` say?  Is it already a `dict`?

Reply:
No. Its a generator

`<class 'praw.models.listing.generator.ListingGenerator'>`

Reply:
OK...  So what value does this algorithm add over what `reddit.subreddit().hot()` already delivers?  If the answer is "not much added value" then this is a how-to-use for praw.

Reply:
@cclauss  I found 2 feasable ways of doing it without `praw` but both of them has some downsights:

1. Scrape reddit using BS4. ( Will be very complicated to scrape data and highly rate limited)
2. Use reddit API directly (Rate limited extremely : like 1 requests per minute)

Which one do you prefer?

I could get the 2nd method working by tomorrow. BS4 might take some time(don't even know if it would be possible)

Reply:
@cclauss  Rewritten using `requests`